fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -1998,9 +1998,18 @@ Optional<std::tuple<Version, Version, std::vector<TLogLockResult>>> TagPartition
 		int safe_range_begin = logSet->tLogWriteAntiQuorum;
 		int new_safe_range_begin = std::min(logSet->tLogWriteAntiQuorum, (int)(results.size() - 1));
 		int safe_range_end = std::max(logSet->tLogReplicationFactor - absent, 1);
-
-		if (!lastEnd.present() || ((safe_range_end > 0) && (safe_range_end - 1 < results.size()) &&
-		                           results[safe_range_end - 1].end < lastEnd.get())) {
+		// The index (in "results" vector) of the recovery version that we will use in the check below
+		// to decide whether to restart recovery not. In "main" we use the version at index
+		// "(safe_range_end - 1)" - this is to minimize the chances of restarting the current recovery
+		// process. With "version vector" we use the version at index "new_safe_range_begin" - this is
+		// because choosing any other version may result in not copying the correct version range to the
+		// log servers in the latest epoch and also will invalidate the changes that we made to the peek
+		// logic in the context of version vector.
+		int version_index =
```

Comment:
do not use camel case format in FDB code. `versionIndex`.

Reply:
I followed the format of the other variables/indexes that were declared in this if-block. Let me know if you still want me to change this.

Reply:
Yes, please

Reply:
Done.